File: WdtcCore/src/main/java/org/wdt/wdtc/utils/UrlUtils.java
package org.wdt.wdtc.utils;

import lombok.SneakyThrows;
import org.wdt.utils.io.IOUtils;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLConnection;

public class UrlUtils {
    public static boolean isNetworkHasThisFile(String url) throws MalformedURLException {
        return isNetworkHasThisFile(new URL(url));
    }

    public static boolean isNetworkHasThisFile(URL url) {
        try {
            URLConnection uc = url.openConnection();
            uc.setConnectTimeout(12000);
            uc.connect();
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    public static String getRedirectUrl(String path) throws IOException {
        HttpURLConnection conn = (HttpURLConnection) new URL(path).openConnection();
        conn.setInstanceFollowRedirects(false);
        conn.setConnectTimeout(5000);
        return conn.getHeaderField("Location");
    }


    @SneakyThrows(MalformedURLException.class)
    public static boolean isOnline() {
        return isNetworkHasThisFile("https://www.bilibili.com");
    }

    public static String getUrlToString(String UrlPath) throws IOException {
        return IOUtils.toString(new URL(UrlPath));
    }

    @SneakyThrows(IOException.class)
    public static void openSomething(Object o) {
        Runtime.getRuntime().exec(new String[]{"cmd.exe", "/c", "start", o.toString()});
    }
}

File: core/core-api/src/main/kotlin/org/wdt/wdtc/core/openapi/utils/URLUtils.kt
package org.wdt.wdtc.core.openapi.utils

import org.wdt.wdtc.core.openapi.manger.networkLinkTimeoutTime
import java.io.IOException
import java.io.InputStream
import java.net.URL


fun URL.isNetworkHasThisFile(): Boolean = try {
	this.openConnection().run {
		connectTimeout = networkLinkTimeoutTime
		connect()
	}
	true
} catch (e: IOException) {
	false
}

fun URL.getRedirectUrl(): String = this.openConnection().apply {
	connectTimeout = networkLinkTimeoutTime
}.getHeaderField("Location")

val isOnline: Boolean
	get() = "https://www.bilibili.com".toURL().isNetworkHasThisFile()


fun openSomething(o: Any) {
	o.toString().also {
		launchScope(it) {
			Runtime.getRuntime().exec(arrayOf("cmd.exe", "/c", "start", it))
			logmaker.info("$it is open")
		}
	}
}

fun URL.newInputStream(): InputStream = openConnection().apply {
	connectTimeout = networkLinkTimeoutTime
	readTimeout = networkLinkTimeoutTime
}.inputStream

fun String.toURL(): URL {
	return URL(this)
}

